### PR TITLE
Fix evm state for getting root chain stakes

### PR DESCRIPTION
**Pull request by ninjaahhh**

### CODE REVIEW

File: quarkchain/cluster/shard_state.py [hunk 1]
```diff
@@ -1888,12 +1888,16 @@ def get_minor_block_by_height(
         return block, posw_info and posw_info._asdict()
 
     def get_root_chain_stakes(
-        self, recipient: bytes, block_hash: bytes
+        self,
+        recipient: bytes,
+        block_hash: bytes,
+        mock_evm_state: Optional[EvmState] = None,  # open for testing
```
Comment (ninjaahhh) [inline quarkchain/cluster/shard_state.py]: cc @syntrust

**Comment by ninjaahhh** [comment]
tested on testnet3, syncing works

File: quarkchain/cluster/shard_state.py [hunk 2]
```diff
@@ -1888,12 +1888,16 @@ def get_minor_block_by_height(
...
     ) -> (int, bytes):
-        h = self.db.get_minor_block_header_by_hash(block_hash)
-        check(h is not None)
-        evm_state = self._get_evm_state_from_height(h.height).ephemeral_clone()
-        evm_state.gas_used = 0
+        meta = self.db.get_minor_block_meta_by_hash(block_hash)
+        check(meta is not None)
+        evm_state = mock_evm_state or self.__create_evm_state(
```
Comment (qizhou) [inline quarkchain/cluster/shard_state.py]: If mock_evm_state is not Null, should we still check meta is not None?

Comment (ninjaahhh) [reply quarkchain/cluster/shard_state.py]: good point. I think we can skip checking meta when mocking, but as for now the tests are not broken, seems no need to change the code

**Pull request merged**